pyglotaran_extras/plotting/plot_traces.py

```diff
@@ -13,14 +13,22 @@ def get_shifted_traces(res, center_λ=None):
     times = traces.coords["time"]
     if center_λ is None:  # center wavelength (λ in nm)
         center_λ = min(res.dims["spectral"], round(res.dims["spectral"] / 2))
-    if "center_dispersion_1" in res:
-        center_dispersion = res.center_dispersion_1  # TODO: clarify against pyglotaran API why _1?
+
+    if "irf_center_location" in res:
+        center_dispersion = res.center_dispersion
+        irf_loc = center_dispersion.sel(spectral=center_λ, method="nearest").item()
+    elif "center_dispersion_1" in res:
+        # legacy compatibility pyglotaran<0.5.0
+        center_dispersion = res.center_dispersion_1
         irf_loc = center_dispersion.sel(spectral=center_λ, method="nearest").item()
     elif "irf_center" in res:
         irf_loc = res.irf_center
     else:
         irf_loc = min(times)
 
+    if hasattr(irf_loc, "shape"):
```

Comment:
Don't take the average of the irf pulses but pass the centers and assume the [0]th element as the default for 'the' irf locaiton.